src/commands/cmd_set.cc

```diff
@@ -283,6 +285,65 @@ class CommandSInter : public Commander {
   }
 };
 
+/*
+ * description:
+ *    syntax:   `SINTERCARD numkeys key [key ...] [LIMIT limit]`
+ *
+ *    limit:    the valid limit is an non-negative integer.
+ */
+class CommandSInterCard : public Commander {
+ public:
+  Status Parse(const std::vector<std::string> &args) override {
+    auto parse_numkey = ParseInt<int>(args[1], 10);
+    if (!parse_numkey) {
+      return {Status::RedisParseErr, errValueNotInteger};
+    }
+    numkeys_ = *parse_numkey;
+
+    // command: for example, SINTERCARD 2 key1 key2 LIMIT 1
+    if (args.size() == numkeys_ + 4 && util::ToLower(args[numkeys_ + 2]) == "limit") {
+      auto parse_limit = ParseInt<int>(args[numkeys_ + 3], 10);
+      if (!parse_limit) {
+        return {Status::RedisParseErr, errValueNotInteger};
+      }
+      limit_ = *parse_limit;
+      if (limit_ < 0) {
+        return {Status::RedisParseErr, errLimitIsNegative};
+      }
+    }
+    return Commander::Parse(args);
+  }
+
+  Status Execute(Server *svr, Connection *conn, std::string *output) override {
+    std::vector<Slice> keys;
+    for (size_t i = 2; i < numkeys_ + 2; i++) {
+      keys.emplace_back(args_[i]);
```

Comment:
It seems there is no check for the boundary of `args_`, e.g. for `SINTERCARD 99 a b` will the `Execute` function just crash on this line?

Reply:
Thank you so much. So many vulnerabilities.

Reply:
Any user input via redis protocol cannot just make kvrocks crash. That is the principal commitment of development.

And you cannot enforce users to always use go-redis wrapped command functions.

Reply:
OK, I understand this and thanks again.